packages/edit-site/src/components/sidebar-dataviews/custom-dataviews-list.js

```diff
@@ -39,6 +55,43 @@ function CustomDataViewItem( { dataviewId, isActive } ) {
 			isActive={ isActive }
 			isCustom="true"
 			customViewId={ dataviewId }
+			suffix={
+				<DropdownMenu
+					icon={ moreVertical }
+					label={ __( 'Actions' ) }
+					toggleProps={ {
+						style: {
+							color: 'inherit',
+						},
+					} }
+				>
+					{ ( { onClose } ) => (
+						<MenuGroup>
+							<MenuItem
+								onClick={ () => {
+									deleteEntityRecord(
+										'postType',
+										'wp_dataviews',
+										dataview.id,
+										{
+											force: true,
+										}
+									);
+									if ( isActive ) {
+										history.push( {
```

Comment:
Haven't tested yet, but if we delete, we should probably use `history.replace`.

Reply:
Good point the code the was updated to use replace.